### DISCUSSION TITLE
2020/04/how-to-create-a-powershell-alias-with-parameters/

### DISCUSSION THREAD

**Discussion by SeanKilleen** [SeanKilleen/seankilleen.github.io#641] [Specific Post Comments]
https://seankilleen.com/2020/04/how-to-create-a-powershell-alias-with-parameters/

**Comment by SeanKilleen** [comment]
## Historical Comments

2020-04-02, Disqus user [Eric Olsson](https://disqus.com/by/ericolsson/):

> Thanks for this. I was exactly looking to be able to do this yesterday (aka, the same day you wrote this blog post).
> 
> The best I found was similar to what you posted--multiple functions that called the parameterized function--each of which could have an alias.
> 
> I just wanted to let you know that somewhere on the Internet, someone else was trying to solve this identical problem. :)

2020-04-02, Me (responding to Eric): 

> Thanks for checking in! :) and I like the tip about a functions that call the parameterized function -- I think folks looking here will find that point to be an additional benefit.
> 
> When writing something up, I often thing "is someone really going to actually find this valuable?" -- so I appreciate the encouragement for future posts as well.

2020-09-30, Disqus user [Maros Kukan]():

> Very useful indeed. In case of string "concatenation" inside a function I had to do slight adjustment in order to get it work.
> 
> 
> function forwardme($LocalPort)
> {
> ssh -i .\.ssh\id_rsa -R "$LocalPort:10.0.0.5:443" maros@10.0.0.4
> }

2021-02-03, Disqus User David Valderrama]():

> ... and then you just pass the parameter value after adding a space after the alias command...

2020-02-04, Me (responding to David):

> Great point, David -- I could certainly include the usage as a last step. Thanks!

**Comment by xathian** [comment]
THANK you for this clear, concise and easy explanation. It seems simple tutorials on setting up PowerShell functions for easy commands+arguments are effectively non-existent online and everything but this was way too over explained and convoluted

**Comment by Magic31** [comment]
Thank you for your support. Very usefull short explanation.

**Comment by cfmx** [comment]
Thank you for this info.
I would only add that $profile file in VSCode is in MyDocuments\PowerShell\Microsoft.VSCode_profile.ps1
Also, after creating my function I added a new alias pointing to my function & voilà!
